### PR TITLE
sinkv2(cdc): introduce EventSortEngineManager to puller

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Issue Number: ref #5928 

### What is changed and how it works?

Add a new interface EventSortEngineManager and introduce it to puller.

Currently EventSortEngineManager coexists with SorterSystem. We will remove SorterSystem later after all sorter implementations are transformed to EventSortEngine.

#### Tests 

 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/manager_test.go [hunk 1]
```diff
@@ -304,7 +304,7 @@ func TestQueryTableCount(t *testing.T) {
 	tableCh := make(chan int, 1)
 	err := m.sendCommand(ctx, commandTpQueryTableCount, tableCh, done)
 	require.Nil(t, err)
-	err = m.handleCommand()
+	err = m.handleCommand(nil)
```
Comment (hi-rustin) [inline cdc/processor/manager_test.go:307]: Can we pass the ctx into this func?

File: cdc/processor/pipeline/puller.go [hunk 2]
```diff
@@ -106,3 +107,52 @@ func (n *pullerNode) start(ctx pipeline.NodeContext,
 	n.cancel = cancel
 	return nil
 }
+
+func (n *pullerNode) start1(ctx pipeline.NodeContext,
```
Comment (hi-rustin) [inline cdc/processor/pipeline/puller.go:111]: How about startWithxxx?(sorter system or sorter engine)

File: pkg/sorter/manager/manager.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+func (f *EventSortEngineManager) Close() (err error) {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	for _, engine := range f.engines {
+		err = multierr.Append(err, engine.Close())
+	}
+	for _, db := range f.dbs {
+		err = multierr.Append(err, db.Close())
+	}
+	return
```
Comment (hi-rustin) [inline pkg/sorter/manager/manager.go:94]:
```suggestion
```

File: pkg/sorter/manager/manager.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	f.mu.Lock()
+	defer f.mu.Unlock()
+
+	engine, exists := f.engines[ID]
+	if !exists {
+		return
+	}
+	delete(f.engines, ID)
+	err = engine.Close()
+	return
```
Comment (hi-rustin) [inline pkg/sorter/manager/manager.go:81]:
```suggestion
	engine, exists := f.engines[ID]
	if !exists {
		return
	}
	delete(f.engines, ID)
	return engine.Close()
```

File: pkg/sorter/manager/manager.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+		exists := false
+		if engine, exists = f.engines[ID]; exists {
+			return engine, nil
+		}
+		// TODO(qupeng): implement pebble engine.
+		engine = nil
+		f.engines[ID] = engine
+		return
+	default:
+		panic("not implemented")
```
Comment (hi-rustin) [inline pkg/sorter/manager/manager.go:66]: Do we need to use` log.Panic` here?

**Review by hi-rustin — COMMENTED** [review]
Thanks!

File: cdc/processor/manager_test.go (hunk 1 above)
Comment (hicqu) [reply cdc/processor/manager_test.go:307]: No, it requires `cdcContext.Context`.

File: pkg/sorter/manager/manager.go (hunk 3 above)
Comment (hicqu) [reply pkg/sorter/manager/manager.go:94]: It can't be removed, `err` should be returned.

**Review by hi-rustin — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**